### PR TITLE
Version bracketing windows cmake workaround, avoid c++17 check

### PR DESCRIPTION
**Pull request by danhoeflinger**
The CMake workaround in `cmake/windows-dpcpp-toolchain.cmake` is causing problems for the newly added c++17 check. 
Version bracketing the workaround to 3.20 >= version < 3.23 based on local testing on windows.   Turning off that check when that workaround is enabled.  

**Can we assume that dpcpp always provides c++17 (on windows)?**
If not, perhaps we can provide an else to manually check the dpcpp version based on known version values?

### CODE REVIEW

**Comment by danhoeflinger** [comment]
I'm not sure why all the compiler flag checks are failing on windows for dpcpp in the CI:

```-- The CXX compiler identification is IntelLLVM 2023.0.0 with MSVC-like command-line
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Miniconda/Library/bin/dpcpp.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring oneDPL 2022.1.0
-- Performing Test _fsycl_option
-- Performing Test _fsycl_option - Failed
-- Build type is release
-- Using parallel policies with TBB backend
icpx: warning: use of 'dpcpp' is deprecated and will be removed in a future release. Use 'icpx -fsycl' [-Wdeprecated]
-- Performing Test _fopenmp_simd_option
-- Performing Test _fopenmp_simd_option - Failed
-- Performing Test _Qopenmp_simd_option
-- Performing Test _Qopenmp_simd_option - Failed
-- Performing Test _qopenmp_simd_option
-- Performing Test _qopenmp_simd_option - Failed
-- Performing Test _openmp_simd_option
-- Performing Test _openmp_simd_option - Failed
-- oneDPL: no effect from enabled ONEDPL_ENABLE_SIMD; unsupported for current compiler
-- Found TBB: TBB::tbb;TBB::tbbmalloc (Required is at least version "2021") found components: tbb tbbmalloc 
-- oneDPL uses oneTBB 2021.8.0
-- Performing Test _fno_fast_math_option
-- Performing Test _fno_fast_math_option - Failed
-- oneDPL: -fno-fast-math is not supported by current compiler
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -54,7 +54,9 @@ endif()
 # Setup the oneDPL library target
 ###############################################################################
 add_library(oneDPL INTERFACE)
-target_compile_features(oneDPL INTERFACE cxx_std_17)
+if(NOT _cmake_windows_compid_workaround)
+    target_compile_features(oneDPL INTERFACE cxx_std_17)
+endif()
```
Comment (dmitriy-sobolev) [inline CMakeLists.txt:59]: Could you apply the same workaround in https://github.com/oneapi-src/oneDPL/blob/main/cmake/templates/oneDPLConfig.cmake.in?

Comment (danhoeflinger) [reply CMakeLists.txt:61]: Yes, thanks.  Done.

**Comment by danhoeflinger** [comment]
https://github.com/oneapi-src/oneDPL/actions/runs/4294512173/jobs/7483639214:
```
[3/360] cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_exe --intdir=test\CMakeFiles\buffer_wrapper.pass.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\Miniconda\Library\bin\dpcpp.exe /nologo test\CMakeFiles\buffer_wrapper.pass.dir\general\buffer_wrapper.pass.cpp.obj  /Qoption,link,/machine:x64 /INCREMENTAL:NO /Qoption,link,/subsystem:console  C:\Miniconda\Library\lib\tbb12.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /link /out:test\buffer_wrapper.pass.exe /implib:test\buffer_wrapper.pass.lib /pdb:test\buffer_wrapper.pass.pdb /version:0.0  && cd ."
FAILED: test/buffer_wrapper.pass.exe 
cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_exe --intdir=test\CMakeFiles\buffer_wrapper.pass.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\Miniconda\Library\bin\dpcpp.exe /nologo test\CMakeFiles\buffer_wrapper.pass.dir\general\buffer_wrapper.pass.cpp.obj  /Qoption,link,/machine:x64 /INCREMENTAL:NO /Qoption,link,/subsystem:console  C:\Miniconda\Library\lib\tbb12.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /link /out:test\buffer_wrapper.pass.exe /implib:test\buffer_wrapper.pass.lib /pdb:test\buffer_wrapper.pass.pdb /version:0.0  && cd ."
LINK: command "C:\Miniconda\Library\bin\dpcpp.exe /nologo test\CMakeFiles\buffer_wrapper.pass.dir\general\buffer_wrapper.pass.cpp.obj /Qoption,link,/machine:x64 /INCREMENTAL:NO /Qoption,link,/subsystem:console C:\Miniconda\Library\lib\tbb12.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /link /out:test\buffer_wrapper.pass.exe /implib:test\buffer_wrapper.pass.lib /pdb:test\buffer_wrapper.pass.pdb /version:0.0 /MANIFEST /MANIFESTFILE:test\buffer_wrapper.pass.exe.manifest" failed (exit code 1) with the following output:
icpx: warning: use of 'dpcpp' is deprecated and will be removed in a future release. Use 'icpx -fsycl' [-Wdeprecated]
icpx: error: no such file or directory: 'kernel32.lib'
icpx: error: no such file or directory: 'user32.lib'
icpx: error: no such file or directory: 'gdi32.lib'
icpx: error: no such file or directory: 'winspool.lib'
icpx: error: no such file or directory: 'shell32.lib'
icpx: error: no such file or directory: 'ole32.lib'
icpx: error: no such file or directory: 'oleaut32.lib'
icpx: error: no such file or directory: 'uuid.lib'
icpx: error: no such file or directory: 'comdlg32.lib'
icpx: error: no such file or directory: 'advapi32.lib'
```
My suspicion is that the issue this PR is meant to fix in cmake has been hiding some other issues on the windows side. Looking back at other runs, I believe the tests have been silently failing by failing the cmake, not generating any tests, and still showing green because the return code of cmake is unchecked.

These errors look like environment setup problems, and may or may not be connected to the previous comment of failing compiler flag checks.

**Pull request closed without merging**